internal/mode/static/state/graph/namespace.go

```diff
@@ -0,0 +1,43 @@
+package graph
+
+import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// buildReferencedNamespaces returns a map of all the Namespace resources in the current clusterState with a label
+// that matches any of the Gateway Listener's label selector.
+func buildReferencedNamespaces(clusterNamespaces map[types.NamespacedName]*v1.Namespace,
+	gw *Gateway,
+) map[types.NamespacedName]*v1.Namespace {
+	referencedNamespaces := make(map[types.NamespacedName]*v1.Namespace)
+	for name, ns := range clusterNamespaces {
+		if checkNamespace(ns, gw) {
+			referencedNamespaces[name] = ns
+		}
+	}
+	if len(referencedNamespaces) == 0 {
+		return nil
+	}
+	return referencedNamespaces
+}
+
+// checkNamespaces returns a boolean that represents whether a given Namespace resource has a label
+// that matches any of the Gateway Listener's label selector.
+func checkNamespace(ns *v1.Namespace, gw *Gateway) bool {
```

Comment:
`checkNamespace` doesn't describe the function very well; it's not clear what we are checking the namespace for. Some alternatives:
- isNamespaceReferenced
- gatewayReferencesNamespace
- namespaceReferencedByGateway

Reply:
I like isNamespaceReferenced! Thanks!